### PR TITLE
Clean up gitignore

### PR DESCRIPTION
**Pull request by Dragorn421**
Remove ancient things from gitignore and gitignoring extracted assets files (now gitignored as part of extracted/)

Here's a one liner to remove all "old files":
`rm -rf .extracted-assets.json *.elf *.z64 *.n64 *.v64 baseroms/gc-eu-mq-dbg/segments/ baseroms/gc-eu-mq/segments/ ; find assets \( -name '*.png' -o -name '*.jpg' \) -delete`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,54 +7,29 @@ __pycache__/
...
 .extracted-assets.json
 extracted/
 
 # Docs
 !docs/tutorial/
 
 # Per-user configuration
-.python-version
+# Use `git config core.excludesFile path/to/my_gitignore_file`
+# if you want to use your own gitignore rules without modifying this file.
```
Comment (cadmic) [inline .gitignore:35]: You can also edit .git/info/exclude

File: .gitignore [hunk 2]
```diff
@@ -7,54 +7,29 @@ __pycache__/
...
-*.ram
 
-# Tool artifacts
-tools/mipspro7.2_compiler/
-tools/overlayhelpers/batchdisasm/output/*
-tools/overlayhelpers/batchdisasm/output2/*
-tools/overlayhelpers/batchdisasm/mipsdisasm/*
+# Tools
+.venv/
 tools/disasm/output/*
```
Comment (cadmic) [inline .gitignore:21]: `tools/disasm/output/*` can probably be removed, it's not used by the current iteration of tools/disasm (according to blame it's from the initial commit)

File: .gitignore [hunk 3]
```diff
@@ -7,54 +7,29 @@ __pycache__/
...
-*.png
-*.jpg
-*.mdli
-*.anmi
-*.obj
-*.mtl
-*.fbx
-!*_custom*
 .extracted-assets.json
 extracted/
```
Comment (cadmic) [inline .gitignore:28]: Maybe this is better with "# Project-specific ignores" like build/?

File: .gitignore [hunk 4]
```diff
@@ -7,54 +7,29 @@ __pycache__/
...
 # Assets
-*.png
-*.jpg
-*.mdli
-*.anmi
-*.obj
-*.mtl
-*.fbx
-!*_custom*
 .extracted-assets.json
```
Comment (cadmic) [inline .gitignore:27]: Is `.extracted-assets.json` kept for convenience because it's mostly harmless?

**Review by cadmic — APPROVED** [review]
Tried it out and I don't currently have anything in my copy that I think should still be gitignored

File: .gitignore (hunk 4 above)
Comment (Dragorn421) [reply .gitignore:27]: Oh I forgot this became extracted/VERSION/assets_extraction_times.json yeah I'll remove that too

File: .gitignore [hunk 5]
```diff
@@ -7,54 +7,29 @@ __pycache__/
...
 extracted/
 
 # Docs
 !docs/tutorial/
```
Comment (cadmic) [inline .gitignore:31]: is `!docs/tutorial/` still necessary now that pngs are not gitignored?

Comment (Dragorn421) [reply .gitignore:31]: I was hesitating to remove due to #2013 just removing the tutorial altogether, but then yeah I guess. didn't quite think it through

**Review by fig02 — APPROVED** [review]

**Pull request merged**